addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerJsonParser.java

```diff
@@ -280,6 +282,77 @@ private static ImageIcon createSvgIcon(URL url) throws Exception {
+        AppPattern appPat;
+        if (json == null) {
+            return Collections.emptyList();
+        }
+        if (json instanceof JSONObject) {
+            for (Object domSelectorObject : ((JSONObject) json).entrySet()) {
+                Map.Entry<?, ?> entry = (Map.Entry<?, ?>) domSelectorObject;
+                for (Object nodeSelectorObject : ((JSONObject) entry.getValue()).entrySet()) {
+                    Map.Entry<?, ?> entr = (Map.Entry<?, ?>) nodeSelectorObject;
+                    if (((Map.Entry<?, ?>) nodeSelectorObject).getValue() instanceof JSONObject) {
+                        for (Object objvalue : ((JSONObject) entr.getValue()).entrySet()) {
+                            Map.Entry<?, ?> ent = (Map.Entry<?, ?>) objvalue;
+                            try {
+                                Map<String, Map<String, Map<String, AppPattern>>> domSelectorMap =
+                                        new HashMap<>();
+                                Map<String, Map<String, AppPattern>> nodeSelectorMap =
+                                        new HashMap<>();
+                                Map<String, AppPattern> value = new HashMap<>();
+                                appPat = this.strToAppPattern(type, (String) ent.getValue());
+                                value.put((String) ent.getKey(), appPat);
+                                nodeSelectorMap.put((String) entr.getKey(), value);
+                                domSelectorMap.put((String) (entry).getKey(), nodeSelectorMap);
+                                list.add(domSelectorMap);
+                            } catch (NumberFormatException e) {
+                                logger.error(
+                                        "Invalid field syntax "
+                                                + ent.getKey()
+                                                + " : "
+                                                + ent.getValue(),
+                                        e);
+                            } catch (PatternSyntaxException e) {
+                                patternErrorHandler.handleError((String) ent.getValue(), e);
+                            }
+                        }
+                    } else {
+                        try {
+                            Map<String, Map<String, Map<String, AppPattern>>> domSelectorMap =
+                                    new HashMap<>();
+                            Map<String, Map<String, AppPattern>> nodeSelectorMap = new HashMap<>();
+                            Map<String, AppPattern> value = new HashMap<>();
+                            appPat = this.strToAppPattern(type, (String) entr.getValue());
+                            value.put((String) entr.getKey(), appPat);
+                            nodeSelectorMap.put((String) entr.getKey(), value);
+                            domSelectorMap.put((String) (entry).getKey(), nodeSelectorMap);
```

Comment:
The extra parentheses can be removed, it would be better to rename `entr` and `entry`.